src/Groups.jl

```diff
@@ -4,4 +4,256 @@
+#   elem_type(::Type{MyGroup})
+#
+# ... and many more of course (among those below and beyond), such as
+#   gens(::Group)
+#   ngens(::Group)
+#   order(::Group)
+#   order(::GroupElem)
+
+
+###############################################################################
+#
+#   Orders of groups and group elements
+#
+###############################################################################
+
+struct InfiniteOrderError{T} <: Exception
+    x::T
+    InfiniteOrderError(g::Union{GroupElem, Group}) = new{typeof(g)}(g)
+end
+
+function Base.showerror(io::IO, err::InfiniteOrderError{T}) where T
+    println(io, "Infinite order exception with ", err.x)
+    print(io, "order will only return a value when it is finite. ")
+    f = if T <: Group
+        "is_finite(G)"
+    elseif T <: GroupElem
+        "is_finiteorder(g)"
+    end
+    print(io, "You should check with `$f` first.")
+end
+
+@doc raw"""
+    order(::Type{T} = BigInt, G::Group) where T
+
+Return the order of $G$ as an instance of `T`. If $G$ is of infinite order,
+`InfiniteOrderError` exception will be thrown.
+
+!!! warning
+
+    Only arbitrary sized integers are required to return a mathematically
+    correct answer.
+"""
+function order(::Type{T}, G::Group) where T
+    throw(NotImplementedError(:order, G))
+end
+
+# if no return type has been specified, default to `BigInt`
+order(G::Group) = order(BigInt, G)
+order(g::GroupElem) = order(BigInt, g)
+
+"""
+    is_finiteorder(g::GroupElem)
+
+Return `true` if `g` is of finite order, possibly without computing it.
+"""
+function is_finiteorder(g::GroupElem)
+    isfinite(parent(g)) && return true
+    throw(
+        InterfaceNotImplemented(
+            :Group,
+            "GroupsCore.is_finiteorder(::$(typeof(g)))",
+        ),
+    )
+end
+
+
+###############################################################################
+#
+#   Properties
+#
+###############################################################################
+
+@doc raw"""
+    is_trivial(G::Group)
+
+Test whether group $G$ is trivial.
+
+The default implementation is based on `is_finite` and `order`.
+"""
+function is_trivial(G::Group)
+    has_gens(G) && return all(isone, gens(G))
+    isfinite(G) && return isone(order(G))
+    return false
+end
+
+###############################################################################
+#
+#   Generating sets
+#
+###############################################################################
+
+@doc raw"""
+    has_gens(G::Group)
+
+Test whether group $G$ has a generating set.
+
+Many algorithms for groups make use of a (finite) generating set. But
+some groups don't have one (e.g. because none has been computed yet, or
+the group is not finitely generated). This test functions allows writing
+algorithms that deal with this case gracefully.
+
+Note that the result of this function can change over time, as side
+effect of a generating set becoming available for a given group.
+"""
+has_gens(G::Group) = true
```

Comment:
The docstring should include the fact, that this is `true` by default and needs to be changed by concrete implementations (if needed).

Reply:
I still think that this either shouldn't be true by default or at least stated so in the docstring

Reply:
I still think that this either shouldn't be true by default or at least stated so in the docstring

Reply:
I've added a note stating what the default is.